src/ConstantSum/ConstantSum.sol

```diff
@@ -58,21 +65,106 @@ contract ConstantSum is PairStrategy {
         )
     {
         ConstantSumParams memory params;
-        (reserves, totalLiquidity, params) =
-            abi.decode(data, (uint256[], uint256, ConstantSumParams));
+
+        (reserves, params) = abi.decode(data, (uint256[], ConstantSumParams));
+        totalLiquidity =
+            computeDeltaLiquidity(reserves[0], reserves[1], params.price);
+
+        if (pool.reserves.length != 2 || reserves.length != 2) {
+            revert InvalidReservesLength();
+        }
 
         internalParams[poolId].price = params.price;
         internalParams[poolId].swapFee = params.swapFee;
+        internalParams[poolId].controller = params.controller;
 
         // Get the trading function and check this is valid
         invariant =
             tradingFunction(reserves, totalLiquidity, abi.encode(params));
 
-        valid = invariant >= 0;
+        valid = invariant >= 0 && invariant <= EPSILON;
 
         return (valid, invariant, reserves, totalLiquidity);
     }
 
+    function validateAllocate(
+        address,
+        uint256 poolId,
+        Pool memory pool,
+        bytes calldata data
+    )
+        external
+        view
+        override
+        returns (
+            bool valid,
+            int256 invariant,
+            uint256[] memory deltas,
+            uint256 deltaLiquidity
+        )
+    {
+        (uint256 deltaX, uint256 deltaY, uint256 minDeltaL) =
+            abi.decode(data, (uint256, uint256, uint256));
+
+        deltaLiquidity =
+            computeDeltaLiquidity(deltaX, deltaY, internalParams[poolId].price);
+        if (deltaLiquidity < minDeltaL) revert InvalidDeltaLiquidity();
```

Comment:
Is there any reason why `ConstantSum.validateAllocate()` was changed to use `deltaX` and `deltaY` as the amount to allocate, and have `minDeltaL` as the slippage parameter?

The parameters for `validateAllocate()` in `ConstantSum` are now inconsistent with other pair strategies, where `deltaL` is the the amount to allocate and `maxDeltaX` and `maxDeltaY` are the slippage parameters. This could be confusing to users - they might call this function thinking that the parameters are the same as the other strategies.

The same goes for `ConstantSum.validateDeallocate()` as well.

Reply:
Yes, so other strategies are computing the deltas given a specific amount of liquidity delta but with the `ConstantSum` we had to change that because we need both deltas to compute `deltaLiquidity`.

Reply:
I see, in that case I don't think this inconsistency is avoidable.